### PR TITLE
go.mod: Upgrade to latest github.com/nspcc-dev/neofs-sdk-go

### PR DESCRIPTION
**Pull request by cthulhu-rider**
untagged, but still

i'm in progress with #2488, maybe addtional SDK fixes will be needed

* based on and blocked by https://github.com/nspcc-dev/neofs-sdk-go/pull/508

### CODE REVIEW

**Comment by carpawell** [comment]
Lets keep it in sync with https://github.com/nspcc-dev/neofs-node/pull/2547. If it is merged first, then #2547 should `Close #2531`. If #2547 is merged first this PR should `Close #2531` (because of https://github.com/nspcc-dev/neofs-sdk-go/pull/515).

File: pkg/local_object_storage/engine/delete.go [hunk 1]
```diff
@@ -121,7 +121,11 @@ func (e *StorageEngine) delete(prm DeletePrm) (DeleteRes, error) {
 
 func (e *StorageEngine) deleteChildren(addr oid.Address, force bool, splitID *objectSDK.SplitID) {
 	var fs objectSDK.SearchFilters
-	fs.AddSplitIDFilter(objectSDK.MatchStringEqual, splitID)
+	if splitID != nil {
```
Comment (roman-khimov) [inline pkg/local_object_storage/engine/delete.go:124]: Is it legitimate to to call `deleteChildren()` with `nil` `splitID`?

Comment (cthulhu-rider) [reply pkg/local_object_storage/engine/delete.go:124]: i left previous behavior untouched. Overall, in current implementation it's safe cuz objects w/o parent ("small") don't appear in `CID_splitid` bucket so they are unprocessed. But it'd be more safe to not call `deleteChildren` at all in this case

**Comment by cthulhu-rider** [comment]
> Lets keep it in sync with https://github.com/nspcc-dev/neofs-node/pull/2547.

u mean they both fix #2531 independetly? if so, then it's ok to have `Closes` notation in both of them imo

File: pkg/local_object_storage/engine/delete.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/local_object_storage/engine/delete.go:124]: I'd make it `splitID objectSDK.SplitID` as well and let the upper layer handle it (not a lot of these layers here).

**Comment by roman-khimov** [comment]
Conflicts

File: pkg/local_object_storage/engine/delete.go (hunk 1 above)
Comment (cthulhu-rider) [reply pkg/local_object_storage/engine/delete.go]: done

**Review by roman-khimov — APPROVED** [review]

**Review by carpawell — APPROVED** [review]

**Pull request merged**